### PR TITLE
Sidebar webb-ui Component 

### LINKED ISSUE #1241: [FEAT] Add Side Navigation Component to Unify Webb dApps
## Overview

As Webb Products continues to expand, there is a need to introduce a side navigation component to create a unified navigation experience across the product ecosystem. This issue aims to provide the necessary information and assets for developing the side navigation component.

### Information Hierarchy

- **Ecosystem:** this section displays and centralizes all Webb’s tools, features, support, and documentation. We could also introduce the vision and value proposition for the Webb Ecosystem here.
- **Account:** this section allows users to manage their "Webb Account," including their account info (public key), shielded balances (available spend notes), address book, help and support, and account settings (including the option to purge account).
- **Hubble:**
    - **Bridge:** private cross-chain asset bridging.
    - **Wrap/Unwrap:** convert webb assets into native assets. 
    - **Explorer:** view shielded pools & data across networks.
    - **Faucet:** access testnet tokens to learn & experiment with Hubble.
- **Tangle Network:**
    - **Faucet:** access tangle testnet tokens,
    - **DKG explorer:** view DKG network status. ****
    - **Homepage:** learn about Tangle network via landing page.

### Tasks

- [x] Review updated design system to access side navigation component & familiarize with different states/variants
- [x] Remove top navigation component in favor of side navigation component on webb dapps (will link updated component in a separate issue)
- [x] Add side navigation components to webb UI kit

### Figma Link (Design System)
https://www.figma.com/file/H43lYnTCOAPjO9kTrrepe1/%F0%9F%95%B8%EF%B8%8F-Design-System?type=design&node-id=5073%3A137026&t=uHKCFH4FOLsP12kS-1

### Screenshots
Screenshot 2023-06-09 at 9 40 34 PM

Image

### Notes
For more information on this task, feel free to access the notion doc below:
https://www.notion.so/hicommonwealth/DOC-Side-Navigation-Evaluation-187a5d958b0746f4b332c5e19e0328bf?pvs=4

### PR DESCRIPTION
**Pull request by devpavan04**
## Summary of changes

- Adds a new `Sidebar` navigation component to webb-ui.

### Proposed area of change

- [x] `apps/bridge-dapp`
- [ ] `apps/stats-dapp`
- [x] `apps/hubble-stats`
- [ ] `apps/webbsite`
- [ ] `apps/faucet`
- [ ] `apps/tangle-website`
- [x] `libs/webb-ui-components`

### Reference issue to close (if applicable)

- Closes #1241 

#### Hubble Bridge

https://github.com/webb-tools/webb-dapp/assets/53374218/c2155cbf-2658-4de2-8c51-63fb197dacf0

#### Hubble Stats

https://github.com/webb-tools/webb-dapp/assets/53374218/2d74cb00-7232-476b-a268-42fcdce33067

#### Sidebar Component Storybook

https://github.com/webb-tools/webb-dapp/assets/53374218/335f8b61-8ee3-4b22-b06b-4ec1ff6fd9b1

#### ThemeToggle Switch Component Storybook

https://github.com/webb-tools/webb-dapp/assets/53374218/e3a9a384-43da-44c8-8b95-e9e7137b099e

-----
### Code Checklist 

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by drewstone** [comment]
Any idea how to fix the CI @devpavan04 ?

File: apps/bridge-dapp/src/components/Header/Header.tsx
Comment (elkremroot) [inline apps/bridge-dapp/src/components/Header/Header.tsx:43]: Why use px values here? This will look tiny on high res monitors

**Review by elkremroot — CHANGES_REQUESTED** [review]

**Comment by elkremroot** [comment]
@devpavan04 when I click directly over toggle, it does not flip: 

image

**Comment by elkremroot** [comment]
apart from ^ LGTM!

**Comment by devpavan04** [comment]
> @devpavan04 when I click directly over toggle, it does not flip:
> 
> image

In the current implementation we need to click on the side we want to switch to. Tried making the whole component clickable resulting to toggle but it is taking a lot longer. I will link an issue to address this later here.

https://github.com/webb-tools/webb-dapp/assets/53374218/13d10640-3261-47ee-8aa9-174e72009536

**Review by elkremroot — APPROVED** [review]

**Review by monaiuu — COMMENTED** [review]
Looks good so far! 

Some comments from me:

- The side nav components are missing state variants (default/hover), as indicated in the figma file. Currently, we have the hover state as default, and perpetually show the open/close controller.
- For the webb logo on top, let's add link to open up webb.tools
- Let's follow the pattern of using accordion option which will allow expanding only one category at once (reference: https://tailwind-elements.com/docs/standard/navigation/sidenav/).

**Comment by devpavan04** [comment]
> Looks good so far!
> 
> Some comments from me:
> 
> * The side nav components are missing state variants (default/hover), as indicated in the figma file. Currently, we have the hover state as default, and perpetually show the open/close controller.
> * For the webb logo on top, let's add link to open up webb.tools
> * Let's follow the pattern of using accordion option which will allow expanding only one category at once (reference: https://tailwind-elements.com/docs/standard/navigation/sidenav/).
> 
> Task for myself:
> 
> * The sublist items currently do not have hover states. I will add that and create a separate issue to address this.

@monaiuu Hover states are missing on the item icons which should be fixed when we address this - https://github.com/webb-tools/webb-dapp/issues/1384

I will add the link to the logo in this PR now.

**Review by vutuanlinh2k2 — CHANGES_REQUESTED** [review]
Made some comments.

**Comment by devpavan04** [comment]
@monaiuu 

- Update bullet points and icon colors.
- Fixed theme switch.


https://github.com/webb-tools/webb-dapp/assets/53374218/4f2366d7-d4aa-400f-8cd6-b600f57933fb

File: apps/bridge-dapp/src/components/Header/Header.tsx
Comment (AtelyPham) [inline apps/bridge-dapp/src/components/Header/Header.tsx:40]: Please remove unused variables.

File: apps/bridge-dapp/src/containers/Layout/Layout.tsx
Comment (AtelyPham) [inline apps/bridge-dapp/src/containers/Layout/Layout.tsx:14]: You should add a type for this.

File: apps/bridge-dapp/src/containers/Layout/Layout.tsx
Comment (AtelyPham) [inline apps/bridge-dapp/src/containers/Layout/Layout.tsx:74]: We should move all the links to `webb-ui-kits` constants.

File: apps/bridge-dapp/src/containers/Layout/Layout.tsx
Comment (AtelyPham) [inline apps/bridge-dapp/src/containers/Layout/Layout.tsx:100]: We should move all the links to `webb-ui-kit` constants.

File: apps/hubble-stats/app/layout.tsx
Comment (AtelyPham) [inline apps/hubble-stats/app/layout.tsx:47]: I'm curious as to why this directs to the development faucet rather than the main faucet page?

File: apps/hubble-stats/app/layout.tsx
Comment (AtelyPham) [inline apps/hubble-stats/app/layout.tsx:65]: Try to use the link inside `webb-ui-kit`, I think we have this link the the `webb-ui-kit`.

File: apps/hubble-stats/app/layout.tsx
Comment (AtelyPham) [inline apps/hubble-stats/app/layout.tsx:74]: Try to use the link inside webb-ui-kit, I think we have this link the the webb-ui-kit.

File: apps/hubble-stats/app/layout.tsx
Comment (AtelyPham) [inline apps/hubble-stats/app/layout.tsx:27]: You should add a type for this variable.

File: apps/hubble-stats/app/layout.tsx
Comment (AtelyPham) [inline apps/hubble-stats/app/layout.tsx:71]: You should add a type for this variable

File: apps/hubble-stats/app/layout.tsx
Comment (AtelyPham) [inline apps/hubble-stats/app/layout.tsx:89]: Use the link inside `webb-ui-kit`.

**Comment by monaiuu** [comment]
A few more items from me:

- Could we disable clickable states for icon items when nav bar is collapsed since they do not open up anything?
https://github.com/webb-tools/webb-dapp/assets/109539270/0a17d775-82c8-4725-bf82-3288360d03fd

- A 'nice-to-have' for opening up the doc site from the bottom of the nav bar is if the system can detect whether the user is on light or dark theme. If the user is initially on light theme, doc site should be light theme, and vice versa. (certainly not crucial)

File: libs/icons/src/index.ts
Comment (AtelyPham) [inline libs/icons/src/index.ts:97]: Could you please clarify why we have two `Tangle` icons present here? Wouldn't it be more efficient to use `<ChainIcon name="tangle" />` instead?

File: libs/webb-ui-components/src/components/LogoWithoutName/LogoWithoutName.tsx
Comment (AtelyPham) [inline libs/webb-ui-components/src/components/LogoWithoutName/LogoWithoutName.tsx:65]: To prevent a UI crash, kindly use `console.error` and default to a valid value.

File: libs/webb-ui-components/src/components/LogoWithoutName/LogoWithoutName.tsx
Comment (AtelyPham) [inline libs/webb-ui-components/src/components/LogoWithoutName/LogoWithoutName.tsx:1]: I'm wondering, why is this necessary instead of using `<ChainIcon name="webb" />`?

File: libs/webb-ui-components/src/components/SideBar/SideBar.tsx
Comment (AtelyPham) [inline libs/webb-ui-components/src/components/SideBar/SideBar.tsx:73]: Put `rel="noopener noreferrer"` to resolve `Potentially unsafe external link ` here.

File: libs/webb-ui-components/src/components/LogoWithoutName/LogoWithoutName.tsx
Comment (AtelyPham) [inline libs/webb-ui-components/src/components/LogoWithoutName/LogoWithoutName.tsx:31]: Please remove unused variable.

File: libs/webb-ui-components/src/components/SideBar/SideBar.tsx
Comment (AtelyPham) [inline libs/webb-ui-components/src/components/SideBar/SideBar.tsx:123]: I think the footer is somewhat similar to the sidebar item, you should use the sidebar item here.

CleanShot 2023-07-05 at 22 56 43@2x

File: libs/webb-ui-components/src/components/SideBar/components/Item.tsx
Comment (AtelyPham) [inline libs/webb-ui-components/src/components/SideBar/components/Item.tsx:22]: Move these types in a `types.ts` file for the library convention.

File: libs/webb-ui-components/src/components/SideBar/components/Item.tsx
Comment (AtelyPham) [inline libs/webb-ui-components/src/components/SideBar/components/Item.tsx:54]: I think this is not really good for accessibility, we should use the anchor tag `<a></a>` for links instead of `window.open` here.

File: libs/webb-ui-components/src/components/SideBar/components/Item.tsx
Comment (AtelyPham) [inline libs/webb-ui-components/src/components/SideBar/components/Item.tsx:1]: You should put this file directly in the `SideBar` folder instead of `SideBar/components` for the library convention. Just like this `libs/webb-ui-components/src/components/Accordion/AccordionButton.tsx`

File: libs/webb-ui-components/src/components/SideBar/components/SubItem.tsx
Comment (AtelyPham) [inline libs/webb-ui-components/src/components/SideBar/components/SubItem.tsx:38]: I think this is not really good for accessibility, we should use the anchor tag `<a></a>` for links instead of `window.open` here.

File: libs/webb-ui-components/src/components/SideBar/utils.ts
Comment (AtelyPham) [inline libs/webb-ui-components/src/components/SideBar/utils.ts:1]: We should use `Link` component for the link instead of `windown.open` here.

File: libs/webb-ui-components/src/components/ThemeToggle/ThemeToggle.tsx
Comment (AtelyPham) [inline libs/webb-ui-components/src/components/ThemeToggle/ThemeToggle.tsx:27]: The toggle function should be placed at the top-level component, (the `<div></div>` in this case), to avoid having non-clickable components.

CleanShot 2023-07-05 at 23 14 41

**Review by AtelyPham — CHANGES_REQUESTED** [review]
Left comments. Let me know if there is any confusion.

For the implementation, you can refer to this file: `libs/webb-ui-components/src/components/Navbar/Navbar.tsx`. The approach we should take is to pass the Link component as a prop and subsequently use that very component for the link.

Additionally, it would enhance the user experience if we could introduce some transition effects when expanding the link.

You can refer to the implementation of these libraries:

- [Ant Design Menu](https://ant.design/components/menu)
- [Daisy UI Menu](https://daisyui.com/components/menu/

File: apps/hubble-stats/app/layout.tsx
Comment (drewstone) [reply apps/hubble-stats/app/layout.tsx:47]: Because it's incorrect :).

**Comment by drewstone** [comment]
Thank you @AtelyPham @monaiuu for great reviews! @AtelyPham amazingly thorough, great work!

**Comment by monaiuu** [comment]
- The control toggle component should use `16px` for icon size, so the control component is `24px` (currently `32px`)

Screenshot 2023-07-06 at 3 37 09 PM

Screenshot 2023-07-07 at 10 30 54 AM


Figma Reference: https://www.figma.com/file/H43lYnTCOAPjO9kTrrepe1/%F0%9F%95%B8%EF%B8%8F-Design-System?type=design&node-id=5082%3A172672&mode=dev

File: apps/hubble-stats/app/layout.tsx
Comment (AtelyPham) [inline apps/hubble-stats/app/layout.tsx:94]: Use metadata here.

Comment (devpavan04) [reply apps/hubble-stats/app/layout.tsx:94]: @AtelyPham This was addressed in the previous [PR](https://github.com/webb-tools/webb-dapp/pull/1394/files) we merged today. Not sure why it hasn't updated here.

**Review by AtelyPham — APPROVED** [review]
Just need small changes, otherwise LG 👍 Please wait for @monaiuu to review the UI before merging the PR.

**Comment by devpavan04** [comment]
@monaiuu 

CleanShot 2023-07-07 at 08 51 54

https://github.com/webb-tools/webb-dapp/assets/53374218/db11b9b2-ed6c-4f0a-b944-87352676e659

https://github.com/webb-tools/webb-dapp/assets/53374218/bb4bf020-d8a9-4229-b581-ea5b0707d80c

**Comment by devpavan04** [comment]
@monaiuu I apologize I missed this one!

https://github.com/webb-tools/webb-dapp/assets/53374218/e18c2660-b8db-4e8a-b753-5742f3b754d0

**Comment by monaiuu** [comment]
> @monaiuu I apologize I missed this one!
> 
>  CleanShot.2023-07-07.at.09.41.37.mp4

No worries. Is it possible to open and close the side nav component by also tapping on any area of the side nav component? Because it might be quite difficult/inaccessible to click directly on the control toggle. 

Let me know your thoughts.

**Comment by devpavan04** [comment]
> > @monaiuu I apologize I missed this one!
> > CleanShot.2023-07-07.at.09.41.37.mp4
> 
> No worries. Is it possible to open and close the side nav component by also tapping on any area of the side nav component? Because it might be quite difficult/inaccessible to click directly on the control toggle.
> 
> Let me know your thoughts.

@monaiuu 

https://github.com/webb-tools/webb-dapp/assets/53374218/1665dbec-b7db-4316-864d-2da08bd80070

**Pull request merged**
